src/pages/QuestionPage.tsx

```diff
@@ -4,25 +4,50 @@ import Question from "../components/Question";
 import Answer from "../components/Answer";
 import { useEffect, useState } from "react";
 import { usePollForGetQuestion } from "../hooks/usePollForGetQuestion";
-import { useAppSelector } from "../state/reduxHooks";
+import { useAppDispatch, useAppSelector } from "../state/reduxHooks";
+import { submitQuestion, setQuestionAnswer } from "../state/questionSlice";
+import { setGotQuestion, setRequestNextQuestion } from "../state/gameSlice";
+import { useSubmitAnswer } from "../hooks/useSubmitAnswer";
+import { useNextQuestion } from "../hooks/useNextQuestion"
 
 export default function QuestionPage() {
 
   const { prompt, answers, index } = useAppSelector(state => state.question);
-  const [answer, setAnswer] = useState<string | undefined>(undefined);
+  const { gotQuestion } = useAppSelector(state => state.game);
+  const { isHost } = useAppSelector(state => state.player);
+  const [answer, setAnswer] = useState<number | undefined>(undefined);
+  const [hasCurrentQuestion, setHasCurrentQuestion] = useState<boolean>(false);
+  const dispatch = useAppDispatch();
 
   usePollForGetQuestion();
+  useSubmitAnswer();
+  useNextQuestion();
 
-  const submitAnswer = () => {
+  const submitAnswer = (event:React.MouseEvent) => {
     // Submit answer to backend
+    console.log("Player submitted answer");
+    dispatch(setQuestionAnswer(answer));
+    dispatch(submitQuestion());
+    dispatch(setRequestNextQuestion(true));
```

Comment:
does setting `RequestNextQuestion` to `true` do anything? it seems like from [this line ](https://github.com/Jhappy77/cpsc559-project-front-end/blob/cb2c585915030795c2299c3d82871078cedebb1c/src/hooks/useNextQuestion.ts#L17) it doesn't since a user can only submit an answer and is not a host

Reply:
It still servers a purpose [here](https://github.com/Jhappy77/cpsc559-project-front-end/blob/cb2c585915030795c2299c3d82871078cedebb1c/src/hooks/usePollForGetQuestion.ts#L37). And I deleted the unused `UseGetQuestion` hook that was causing the confusion.